engine/src/main/java/com/datatorrent/stram/StreamingContainerManager.java

```diff
@@ -2123,7 +2122,6 @@ public void updateRecoveryCheckpoints(PTOperator operator, UpdateCheckpointsCont
     for (PTOperator groupOper : groupOpers) {
       // checkpoint frozen during deployment
       if (!pendingDeploy || ctx.recovery) {
-        // remove previous checkpoints
         Checkpoint c1 = Checkpoint.INITIAL_CHECKPOINT;
```

Comment:
The INITIAL_CHECKPOINT seems to be included into the purge list. If it is excluded later, how this exclusion happens?